# Cascade plot from the meta-analysis results

## Output
https://drive.google.com/drive/folders/1fGlnrj8Zu1Pox3fmxmFKdcbNVWiN9vhd

## Lists of input files

- `imp_hits.lst`: list of meta-analysis results files from the meta-analysis on imputed dataset
- `cascade.input.files.tsv`: a table of the meta-analysis results generated by `1_cascade.input.files.ipynb`

## Notebooks
- `cascade_helper.R`: a set of helper functions used in the notebook
- `1_cascade.input.files.ipynb`: generate the table of input files
- `2_cascade_array.hits.ipynb`: aggregate the meta-analysis results for the array, apply p-value thresholding, and annotate the variants.
- `3_cascade_array.hits.plots.ipynb`: generate cascade plots for the protein-truncating variants and protein-altering variants on genotyping array
- `4_cascade_imp.hits.ipynb`: aggegate the meta-analysis results for the imputed dataset and apply p-value thresholding
- `5_cascade_imp.hits.plots.ipynb`: generate cascade plots for the imputed dataset.

## Methods 

### Array
- We apply p-value threshold and focus on associations on LD-independent and non-MHC variant. We used the variant annotation file in the lab.

### Imputed dataset
- We first apply p-value thresholds on "filtered-hits".
- We plot non-MHC autosomal variants.

#### ToDo
- This procedure still includes some variants on imputed dataset that are in LD with array dataset (because the `mfs.R` script does not remove variants that are in LD with array variants.)
- The proper steps should be:
  1. check if the variants are represented on array
  2. run clumping
  3. remove clumped variants if the lead SNPs or other SNPs in the same clump have array variants
  4. run CM-based filtering as in `mfs.R`
- Link: https://github.com/rivas-lab/biomarkers/blob/master/filtration/mfs.R
